MicroPython/src/controller/main_controller.py

```diff
@@ -156,23 +157,25 @@ def process_event(self, event: MainControllerEvent) -> None:
             logging.debug("GETTING SENSOR DATA")
 
             self.data_acquisitor.acquire_temp_humi()
-            self.got_sensor_data = True
+            if not any([FAILED_TO_MEASURE_VALUE in val for (val,) in self.data_acquisitor.data.values()]):
+                self.got_sensor_data = True   
 
         elif event.event_type == MainControllerEventType.PUBLISH_DATA:
             logging.debug("WAITING FOR GOT SENSOR DATA")
-            while not self.got_sensor_data:
-                pass
```

Comment:
I am not sure if this while loop is not needed anyway. There might be a situation when sensor is trying to gather data but it needs some more time (as the code execution may be faster)...Not sure if this is what happens but there is such possibility. Maybe wait for a while instead of waiting indefinitely for sensor to gather data?